### DISCUSSION TITLE
How can I get 'Accuracy, True Positive, False Negative, True Negative, and False Positive from yolov5?

### DISCUSSION THREAD

**Discussion by soleymany** [ultralytics/yolov5#6315] [Q&A]
Hi to all, I got  'P, R, and mAP' from yolov5 **after I did my test**, now I want to get these metrics **after my test** : A, TP, FN, TN, and FP.
I don't know how I can get these metrics **after my test**  ('**A, TP, FN, TN, and FP**')?

**Comment by glenn-jocher** [comment]
@soleymany I think Accuracy only applies to classification tasks.

The others you can collect here on L254, minus TN which doesn't make any sense in this context as it'll always be in the tens of thousands.
https://github.com/ultralytics/yolov5/blob/436ffc417ac2312de18287ddc4f87bdc2f7f5734/val.py#L251-L257

**Reply by soleymany** [reply]
@glenn-jocher  Thanks, I already saw these lines in the **val.py** file, but I do not know **what commends to add to my main program to give me these values in the output?**

**Reply by glenn-jocher** [reply]
@soleymany python uses `return` statement to provide output from a function
